docker-compose.yml

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2021 LEIDOS.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+version: "2"
+services:
+  co-simulation:
+    privileged: true
+    image: ns-3-integration:latest
+    network_mode: host
+    environment:
+      - DISPLAY=${DISPLAY}
+      - QT_X11_NO_MITSHM=1
+    volumes:
+      - /tmp/.X11-unix:/tmp/.X11-unix
+    deploy:
+      resources:
+        reservations:
+          devices:
+            - capabilities: [gpu]
+    command: bash -c "export NS3_HOME='/ns-3-integration/bin/fed/ns3' && cd /ns-3-integration/ && ./mosaic.sh -s ${map}"
+
+  carla-carma-integration:
+    image: carla-carma:latest
+    network_mode: host
+    environment:
+      - ROS_IP=127.0.0.1
+    command: bash -c "sleep 10; export PYTHONPATH=$PYTHONPATH:/home/PythonAPI/carla-0.9.10-py2.7-linux-x86_64.egg &&
```

Comment:
I'd recommend adding much of this to the run.sh. Is the sleep 10 command to allow for roscore to startup or is it to wait specifically for CARLA? Either way, there's a script we use in the carma-base repo called `wait-for-it.sh` that handles this a bit more elegantly. It'll sleep until a specific port (the roscore port 11311 or whatever port CARLA server is listening on) becomes available for connection. I'd recommend looking at that script in the CARMA base repo.

Reply:
The `sleep 10` command is for waiting MOSAIC CARLA bridge to load the right scenario. Currently the CARLA server and switching scenario are handling by co-simulation tool which is not able to get the information about when the right scenario has been loaded.